src/transformers/models/clip/modeling_clip.py

```diff
@@ -750,13 +753,27 @@ def forward(
         last_hidden_state = encoder_outputs[0]
         last_hidden_state = self.final_layer_norm(last_hidden_state)
 
-        # text_embeds.shape = [batch_size, sequence_length, transformer.width]
-        # take features from the eot embedding (eot_token is the highest number in each sequence)
-        # casting to torch.int for onnx compatibility: argmax doesn't support int64 inputs with opset 14
-        pooled_output = last_hidden_state[
-            torch.arange(last_hidden_state.shape[0], device=last_hidden_state.device),
-            input_ids.to(dtype=torch.int, device=last_hidden_state.device).argmax(dim=-1),
-        ]
+        if self.eos_token_id == 2:
```

Comment:
The if/else condition should be enough.

I don't think there is any user changing this from `2` to anything that is not `vocab_size - 1` (the `vocab_size` of the original clip tokenizer)

Reply:
This looks good, but I'm always a bit afraid of hardcoding values like this, as we never know how the users may have modified the config. I will defer to @amyeroberts for this :)

Reply:
We're all a bit afraid BUT I think it's OK 😬 

I think you made a convincing point before, @patil-suraj. This token isn't used by the model anywhere (yet), and `eos_token_id=2` is wrong. There's three possible cases: 

* `eos_token_id=2` - then the old behaviour is still preserved
* `eos_token_id=vocab_size - 1` - then the model behaviour is still preserved (and now correct)
* `eos_token_id=something_else`. As it's not used by the model, it doesn't really make sense to update this unless the tokenizer is also updated with a new `eos_token_id` and then finetuned. There's a small chance this has happened, however I think we would have had this `eos_token_id` inconsistency flagged beforehand in this case and the previous pooling behaviour would have been incorrect. 

I propose waiting until after the branch is cut, and then merging. WDTY?

Reply:
I can wait the branch cut 😄 !

> eos_token_id=something_else

The chance is very small, so I won't spend time to change it to `if eos_token_id < default config value (the now `49407`)`.

(as this needs to inspect the __init__ argument of a class)